cmd/generate.go

```diff
@@ -0,0 +1,45 @@
+package cmd
+
+import (
+	"dynamic-buildkite-template/generator"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	// cobra.OnInitialize()
+	generateCmd.PersistentFlags().String("trivyPlugin", "v1.18.2", "provide trivy plugin version")
+	generateCmd.PersistentFlags().String("shellPlugin", "", "provide shell plugin version")
+	generateCmd.PersistentFlags().Bool("ignoreUnfixed", true, "provide if unfixed items are to be ignored")
+	generateCmd.PersistentFlags().String("skipFiles", "cosign.key", "provide files to be skipped in trivy plugin")
+	generateCmd.PersistentFlags().String("shellCheckFiles", "script.sh", "provide files to be checked by the shell plugin")
+
+	generateCmd.PersistentFlags().StringArray("severity", []string{"CRITICAL", "HIGH"}, "provide the severity")
+	generateCmd.PersistentFlags().StringArray("securityChecks", []string{"config", "secret", "vuln"}, "provide the security checks")
+}
+
+func Execute() error {
+	return generateCmd.Execute()
+}
+
+var generateCmd = &cobra.Command{
+	Use:   "generate",
+	Short: "Generates trivy step for the provided options",
```

Comment:
Is this just for trivy? It looks like it has a shell plugin too?

Reply:
Both trivy and shell plugin are there as parameter and conditions have been added to the template. Done